lib/lhm/entangler.rb

```diff
@@ -95,6 +101,40 @@ def revert
 
     private
 
+    def special_origin?
+      TABLES_WITH_LONG_QUERIES.include? @origin.name
+    end
+
+    def kill_long_running_queries_on_origin_table!
+      return unless ENV['LHM_KILL_LONG_RUNNING_QUERIES'] == 'true'
+      3.times do
+        long_running_queries(@orogin.name).each do |id, query, duration|
+          Lhm.logger.info "Action on table #{table_name} detected; killing #{duration}-second query: #{query}."
+          @connection.execute("KILL #{id}")
```

Comment:
Just a side-note that only the owning user can kill processes. 99% of the time that's fine because the queries will be run by the same user as the LHM migration, but how do we handle long queries run by e.g. the read-only user?

Logging the result of the `KILL` might be enough here. We can always manually kill things in the rare instances the scripted `KILL` doesn't work.

Reply:
I do not have a good answer for this. If like `vangogh-2015-10` has a long running query but `lhm` is using user `heroku-2015-10`, what do we do?
1. I don't know if that is a real concern. i.e. do we actually have long running queries from non `heroku-2015-10` ?
2. I don't think master user can even kill other users' queries. I tried `live` on snapshotDB to kill other queries and got an error.